### DISCUSSION TITLE
Getting Started with Seeed Studio XIAO RA4M1 | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#1597] [Q&A]
# Getting Started with Seeed Studio XIAO RA4M1 | Seeed Studio Wiki

XIAO RA4M1 - Smallest Dev Borad Based on 32-bit Renesas RA4M1, Arduino IDE-Ready, 19 GPIOs, RGB LED, CAN BUS, USB 2.0

https://wiki.seeedstudio.com/getting_started_xiao_ra4m1/

**Comment by artmekab** [comment]
Any example with CAN? What library to use etc.

**Comment by mkoch44** [comment]
On which pin is the 12-bit DAC? Is there a small code example for the DAC?

**Reply by mkoch44** [reply]
The DAC is on pin A0.

**Comment by mkoch44** [comment]
1. Which options are possible for the analogReference() function, and to which pins must an external reference voltage be connected?
The default configuration does obviously use the 3.3V supply as reference, and there is a lot of noise in the ADC measurements.
2. Obviously the DAC does also use 3.3V as reference by default. Is it possible to use another internal or external reference for the DAC?

**Reply by mkoch44** [reply]
It's disappointing that there is so much noise in the ADC values. The SAMD21 module works _much_ better in the same circuit.

**Comment by Jabadabado** [comment]
Is there an STL file for this board? (Maybe any other board has the same dimensions? )

**Comment by Lenigan96** [comment]
Any files for EAGLE?

**Reply by nfs0619** [reply]
Thanks for reaching out! The files we can provide are limited to those available in the Resources section of the documentation [here](https://wiki.seeedstudio.com/getting_started_xiao_ra4m1/#resources). If you have any specific questions about using the product, feel free to contact our technical support team at **[techsupport@seeed.io](mailto:techsupport@seeed.io)**. They will be happy to assist you further.

**Comment by wavejumper00** [comment]
Which pins are the CAN pins?  It is shown twice in the pinouts - the "front" side pinout shows D9 as CRX0 and D10 as CTX0. And on the “back” side pinout, it shows D17 as CRX0 and D18 as CTX0.  I have doubts that it is in really in two places.

Also, please provide instructions on how to use this native can controller in the RA4M1 with only an external transceiver.  The existing examples seem to show the seeed studio breakout board which has SPI connection to external controller which then goes to external transceiver.

I want to use the native CAN TX/RX to connect to only an external transceiver.
Thank you.

**Reply by nfs0619** [reply]
For clarification on the CAN pins and instructions on using the native CAN controller with an external transceiver, we recommend contacting our technical support team at **[techsupport@seeed.io](mailto:techsupport@seeed.io)**. They will be able to provide you with more detailed guidance on your setup and usage.

**Comment by Lenigan96** [comment]
Hey everyone! I have a new question... I have little project and I was wondering... Is it possible to o connect the ESP3253 (Master) with the RA4M1 (slave)?

**Reply by nfs0619** [reply]
It’s definitely possible to connect the ESP32S3 (master) with the RA4M1 (slave) for communication. However, you'll likely need to design the circuit yourself to match the interfaces and communication protocols between the two. Make sure that both can properly connect and communicate.

**Comment by dbarwacz** [comment]
I'm seeing conflicting information on the internet when it comes to powering up this board. 
I assumed there are 3 ways to power this board: 
1. with 5V through USB 
2. with 3.3V directly to 3.3V pin.
3. with 5V directly to 5V pin (labeled as Power Input/Output on the pinout)

The option 3 does not work for me though - I measured 5V on the pins, but the board does not boot. When connected through USB it does boot. 

Can you please confirm if 5V pin is both power input/output and if there are any extra steps required to power the board with direct 5V connection?

**Reply by Lukilyy** [reply]
Hello, regarding the power supply and boot issue:
1. Please check your code to see if there is a while(!Serial) statement. If this is present, it may prevent the board from booting properly when powered by a battery or external source without a USB serial connection.
2. To ensure proper charging and operation, please make sure you are using a 5V or 3.7V power supply.

**Comment by moritz1000** [comment]
The Pin-Map table: https://wiki.seeedstudio.com/getting_started_xiao_ra4m1/#pin-map
**is missing pin P015/22**, that is shown in the screenshot (battery management logic): https://files.seeedstudio.com/wiki/XIAO-R4AM1/img/XIAO_RA4M1_back_pinout.png
Is this intentional?

e.g. over at https://wiki.seeedstudio.com/xiao_rp2350_arduino/#pin-map you intentionally mention the ADC_BAT_EN pin.

**Reply by grilliiiiii** [reply]
> The Pin-Map table: https://wiki.seeedstudio.com/getting_started_xiao_ra4m1/#pin-map **is missing pin P015/22**, that is shown in the screenshot (battery management logic): https://files.seeedstudio.com/wiki/XIAO-R4AM1/img/XIAO_RA4M1_back_pinout.png Is this intentional?
> 
> e.g. over at https://wiki.seeedstudio.com/xiao_rp2350_arduino/#pin-map you intentionally mention the ADC_BAT_EN pin.

Thank you for pointing this out so carefully. Our current presentation may have caused some misunderstanding. We will correct it as soon as possible and clearly indicate the P400 and P015 pins in the table.
